blocks/inspector-controls/checkbox-control/index.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * WordPress dependencies
+ */
+import { withInstanceId } from 'components';
+
+/**
+ * Internal dependencies
+ */
+import BaseControl from './../base-control';
+import './style.scss';
+
+function CheckboxControl( { label, heading, checked, instanceId, onChange, ...props } ) {
+	const id = 'inspector-checkbox-control-' + instanceId;
+	const onChangeValue = ( event ) => onChange( event.target.value );
+
+	return (
+		<BaseControl label={ heading } id={ id }>
+			<label className="blocks-checkbox-control__label" htmlFor={ id }>
```

Comment:
Missed this before. We shouldn't combine approaches of implicit and explicit labelling, between `htmlFor` and nesting input as a child of the label. From previous conversations with @afercia , `htmlFor` should be our preferred approach as it is better supported by screen readers.

https://github.com/WordPress/gutenberg/pull/527#discussion_r113741025